pkg/sloop/ingress/kubewatcher.go

```diff
@@ -306,14 +306,26 @@ func (i *kubeWatcherImpl) processUpdate(kind string, obj interface{}, watchResul
 	glog.V(99).Infof("processUpdate: obj json: %v", resourceJson)
 
 	kubeMetadata, err := kubeextractor.ExtractMetadata(resourceJson)
+	EventInfo, err1 := kubeextractor.ExtractEventInfo(resourceJson)
+	InvolvedObject, err2 := kubeextractor.ExtractInvolvedObject(resourceJson)
 	if err != nil || kubeMetadata.Namespace == "" {
 		// We are only grabbing namespace here for a prometheus metric, so if metadata extract fails we just log and continue
 		glog.V(2).Infof("No namespace for resource: %v", err)
 	}
+	if err1 != nil {
+		glog.V(2).Infof("Extract event info: %v", err1)
+	}
+	if err2 != nil {
+		glog.V(2).Infof("Error occured while extracting Involved Object Info: %v", err2)
+	}
+	if enableGranularmetrics && kind == "Event" {
+		metricIngressGranularKubewatchcount.WithLabelValues(kind, watchResult.WatchType.String(), kubeMetadata.Namespace, kubeMetadata.Name, InvolvedObject.Kind, EventInfo.Reason, EventInfo.Type).Inc()
```

Comment:
line 309, 310 , 315 - 320 should be moved inside the if condition at line 321, as they are not needed otherwise

Reply:
Noted and resolved @sana-jawad